From f2e41a309b7735af3eea60fff64dfc81a8cfbbb3 Mon Sep 17 00:00:00 2001
From: Peter Yin <peter.yin@quantatw.com>
Date: Mon, 3 Nov 2025 21:59:51 +0800
Subject: [PATCH 3/4] configuration: harma: add PDB power monitor calibration

Add calibration data for the first and second PDB power
monitor components.

Change-Id: If392d2db6194f74400aade0396baf2cc5f2fe8da
Signed-off-by: Peter Yin <peter.yin@quantatw.com>
---
 configurations/meta/minerva/minerva_pdb.json         | 7 ++++++-
 configurations/meta/minerva/minerva_pdb_hsc_xdp.json | 7 ++++++-
 schemas/legacy.json                                  | 2 +-
 3 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/configurations/meta/minerva/minerva_pdb.json b/configurations/meta/minerva/minerva_pdb.json
index 6b6454d..c3bcc17 100644
--- a/configurations/meta/minerva/minerva_pdb.json
+++ b/configurations/meta/minerva/minerva_pdb.json
@@ -29,8 +29,13 @@
             ],
             "Type": "INA238",
             "curr1_Name": "PDB_P48V_CURR_A",
+            "curr1_Offset": 0.15,
+            "curr1_Scale": 1041,
             "in1_Name": "PDB_P48V_VOLT_V",
-            "power1_Name": "PDB_P48V_PWR_W"
+            "in1_Offset": 0.888,
+            "power1_Name": "PDB_P48V_PWR_W",
+            "power1_Offset": 8.39,
+            "power1_Scale": 1020408
         },
         {
             "Address": "0x44",
diff --git a/configurations/meta/minerva/minerva_pdb_hsc_xdp.json b/configurations/meta/minerva/minerva_pdb_hsc_xdp.json
index bcc648e..9624934 100644
--- a/configurations/meta/minerva/minerva_pdb_hsc_xdp.json
+++ b/configurations/meta/minerva/minerva_pdb_hsc_xdp.json
@@ -29,8 +29,13 @@
             ],
             "Type": "SQ52206",
             "curr1_Name": "PDB_P48V_CURR_A",
+            "curr1_Offset": 0.15,
+            "curr1_Scale": 1030,
             "in1_Name": "PDB_P48V_VOLT_V",
-            "power1_Name": "PDB_P48V_PWR_W"
+            "in1_Offset": 0.852,
+            "power1_Name": "PDB_P48V_PWR_W",
+            "power1_Offset": 8.857,
+            "power1_Scale": 1020408
         },
         {
             "Address": "0x49",
diff --git a/schemas/legacy.json b/schemas/legacy.json
index b74856c..87448b7 100644
--- a/schemas/legacy.json
+++ b/schemas/legacy.json
@@ -829,7 +829,7 @@
                 "power(1[0-7]|[1-9])?_Max$": {
                     "$ref": "#/$defs/Types/MinMaxType"
                 },
-                "(power|curr|iout|pout|pin)(1[0-7]|[1-9])?_Offset$": {
+                "(in|power|curr|iout|pout|pin)(1[0-7]|[1-9])?_Offset$": {
                     "$ref": "#/$defs/Types/OffsetType"
                 }
             },
-- 
2.47.3

